nml/expression/array.py

```diff
@@ -31,3 +31,9 @@ def __str__(self):
 
     def reduce(self, id_dicts = [], unknown_id_fatal = True):
         return Array([val.reduce(id_dicts, unknown_id_fatal) for val in self.values], self.pos)
+
+    def collect_references(self):
+        refs = []
+        for v in self.values:
+            refs += v.collect_references()
+        return refs
```

Comment:
```suggestion
        return [v.collect_references() for v in self.values]
```
would be slightly more pythonic

Reply:
more like sum((v.collect_references() for v in self.values), [])
or itertools.chain.from_iterable for max effeciency